client/src/components/Button.tsx

```diff
@@ -0,0 +1,20 @@
+type ButtonProps<T> = {
+  action: (args?: T) => void;
+  children: React.ReactNode;
+  className?: string;
+};
+
+export default function Button<T>({
+  action,
+  children,
+  className,
+}: ButtonProps<T>) {
+  function handleClick() {
+    action();
```

Comment:
The name of this function is a little vague - what do you think about making it more specific?

Reply:
i was thinking of this as a generic shared component that's why I keep the function name general. it's naming i've seen in other projects so it's not unusual. questionable if the component is really necessary since it's a pretty thin abstraction over the button element but gonna leave it and if you all decide to take it out later that's cool.

Reply:
Sounds good to me!